### PR TITLE
Enable stack protector on windows / macos and disable for hppa.

### PR DESCRIPTION
**Pull request by rockihack**
This pull request enables stack-protector for windows / macos
and disables it for hppa builds (currently not supported).

I tested the windows and macos builds for a couple of weeks and it works just fine.
Hppa build is untested!

Follow up of: https://github.com/owncloud/client/pull/6040
Fixes: https://github.com/owncloud/client/issues/6211

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -8,30 +8,29 @@ if(NOT TOKEN_AUTH_ONLY)
...
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--nxcompat -Wl,--dynamicbase")
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--nxcompat -Wl,--dynamicbase")
-elseif(UNIX AND NOT APPLE)
-  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.9)
-      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-strong")
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector-strong")
-    else()
+if(NOT MSVC)
+  if(NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "^(parisc|hppa)"))
```
Comment (ckamm) [inline src/CMakeLists.txt]: This if used to have also a `AND NOT APPLE`. @guruz Will setting these compiler flags be ok on osx?

Comment (guruz) [reply src/CMakeLists.txt]: Yep should work, default compiler on macOS is clang

Comment (hefee) [reply src/CMakeLists.txt]: for sure you need also an |alpha cause also alpha platform is affective.

**Comment by guruz** [comment]
We reverted this partly for 2.4.0 in 298f1ab57000afd7acb8c0ddbddca8a1fc6cd92a because our packages don't ship libssp-0.dll which is suddenly needed when enabling the compile flag.

**Pull request merged**